### ISSUE TITLE
Breakpoints Not Working

### ISSUE THREAD

**Issue by appymedia** [vshaxe/vshaxe-debugadapter#12]
Hi,

I'm new to Haxe and using it with VS Code so excuse me if I'm doing something stupid but I'm unable to get breakpoints to work with the Haxe Debug extension.

OSX 10.12.4
VS Code 1.11.2
Haxe Debug Extension 1.1.1
Flash Player Debugger 25.0.0.127
JDK 8u121

I'm getting the following output when setting a breakpoint (by using "receiveAdapterOutput": true in my tasks.json) :-

>DebugAdapter: {"command":"setBreakpoints","arguments":{"source":{"path":"/Users/peter/Desktop/FlixelApp/source/PlayState.hx"},"lines":[14],"breakpoints":[{"line":14}],"sourceModified":false},"type":"request","seq":34}
DebugAdapter: sendResponse: {
	seq : 0, 
	type : response, 
	request_seq : 34, 
	command : setBreakpoints, 
	success : false, 
	message : TypeError: Cannot read property 'split' of undefined
    at Function.StringTools.replace (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:155:10)
    at Object.getKey (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:1847:25)
    at Object.execute (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:1733:22)
    at DebugSession.setBreakPointsRequest (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:305:11)
    at DebugSession.dispatchRequest (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/node_modules/vscode-debugadapter/lib/debugSession.js:312:22)
    at DebugSession.dispatchRequest (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:262:50)
    at DebugSession.ProtocolServer._handleData (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/node_modules/vscode-debugadapter/lib/protocol.js:78:38)
    at Socket.<anonymous> (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/node_modules/vscode-debugadapter/lib/protocol.js:23:60)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7), 
	body : {
		error : {
			id : 1104, 
			format : {_stack}, 
			variables : {
				_exception : Cannot read property 'split' of undefined, 
				_stack : TypeError: Cannot read property 'split' of undefined
    at Function.StringTools.replace (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:155:10)
    at Object.getKey (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:1847:25)
    at Object.execute (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:1733:22)
    at DebugSession.setBreakPointsRequest (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:305:11)
    at DebugSession.dispatchRequest (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/node_modules/vscode-debugadapter/lib/debugSession.js:312:22)
    at DebugSession.dispatchRequest (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/bin/fdb_adapter.js:262:50)
    at DebugSession.ProtocolServer._handleData (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/node_modules/vscode-debugadapter/lib/protocol.js:78:38)
    at Socket.<anonymous> (/Users/peter/.vscode/extensions/vshaxe.haxe-debug-1.1.1/node_modules/vscode-debugadapter/lib/protocol.js:23:60)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
			}, 
			sendTelemetry : true
		}
	}
}

Am I doing something wrong or is this a bug?  Let me know if you need any more info.

**Comment by Gama11** [comment]
I can reproduce this, definitely a bug - thanks for reporting! It seems this broke with VSCode 1.11.0, still works fine in 1.10.0. Here's a comparison of `setBreakpoints` requests in the two versions:

```diff
{
    "command": "setBreakpoints",
    "arguments": {
        "source": {
            "path": "c:\\HaxeToolkit\\haxe\\lib\\flixel-demos\\git\\Platformers\\Mode\\source\\EnemyBullet.hx",
-            "name": "EnemyBullet.hx"
        },
        "lines": [
            22
        ],
        "breakpoints": [
            {
                "line": 22
            }
        ],
        "sourceModified": false
    },
    "type": "request",
    "seq": 3
}
```

The `arguments.source.name` field isn't there anymore, but vshaxe-debugadapter assumes it is always present (no null-checks or anything), so it crashes.

**Comment by Gama11** [comment]
This might be a VSCode bug, see https://github.com/Microsoft/vscode-debugadapter-node/issues/109. However, considering that `name` is part of `path`, we can probably work around it.

**Comment by appymedia** [comment]
Confirm the same here, rolled back VS Code to 1.10.2 and the super awesome power of Haxe Debug is working great :)  Many thanks for the fast response.  If I get to a point my skills allow me to fix and help I will.  Thanks again for your time, appreciated greatly.  Not sure I've ever been so spoiled with the combo of VS Code debugging and the HaxeFlixel Debug overlay!

**Comment by Gama11** [comment]
I just published 1.1.2 - with that, setting breakpoints should work again even with the March release of VSCode. Let me know if it works for you! :)

**Comment by appymedia** [comment]
Working perfect, no problems here with VS Code 1.11.2 and the update to 1.1.2 Haxe Debug.  Thanks again :)
